src/server/search/doc_accessors.cc

```diff
@@ -107,6 +137,9 @@ SearchDocData JsonAccessor::Serialize(search::Schema schema) const {
   return out;
 }
 
+thread_local absl::flat_hash_map<std::string, std::unique_ptr<JsonAccessor::JsonPathContainer>>
+    JsonAccessor::path_cache_;
+
```

Comment:
I've chosen to use this tl cache instead of polluting the core json parts with json stuff. Alternatively we can choose to either wire json to the core or use just a generic member like `std::any access_helper_` to get by without a type